### ISSUE TITLE
shell-out filter idea

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#3755]
A new filter plugin which can shell out to any external program and uses a stdin/stdout json protocol to pass events in & out of that external bin which executes the actual filtering.

this stdin/stdout json protocol could be used in a streaming way to avoid having to spawn the external bin  on every invocation.

**Comment by andrewvc** [comment]
I really really really like this idea
:+1:

**Comment by MaxJust** [comment]
Yes cool (shell and sockets) and  It will be more cool to make msgpack serializator compatibility.
